Bio/Blast/NCBIWWW.py

```diff
@@ -276,26 +278,65 @@ def qblast(
 
         request = Request(url_base, message, {"User-Agent": "BiopythonClient"})
         handle = urlopen(request)
-        results = handle.read().decode()
+        results = handle.readlines()
 
         # Can see an "\n\n" page while results are in progress,
         # if so just wait a bit longer...
-        if results == "\n\n":
+        if results == [b"\n\n"]:
             continue
-        # XML results don't have the Status tag when finished
-        if "Status=" not in results:
-            break
-        i = results.index("Status=")
-        j = results.index("\n", i)
-        status = results[i + len("Status=") : j].strip()
-        if status.upper() == "READY":
+
+        # Iterate over the results to know if the request is complete.
+        is_done = False
+        for line_no, line in enumerate(results):
+            # Has a Status tag and Status is READY
+            if b"Status=" in line:
+                status = results[line_no + 1].strip()
+                if status.upper() == "READY":
+                    is_done = True
+                    break
+        # Or, XML results don't have the Status tag when finished
+        else:
+            is_done = True
+
+        if is_done:
             break
-    return StringIO(results)
+    return _sanitize_qblast_xml(results)
```

Comment:
Do we really need to write a temp file? How about something like this (untested):

```python

return StringIO("".join(line for line in results if line and line != "CREATE_VIEW\n"))
```

Reply:
You could just write each line directly to the StringIO, instead of writing to the tempfile first.

Reply:
Isn't this XML? XML shouldn't be decoded, but passed to the parser as bytes.

Reply:
I saw the `.decode()` call in the original code and thought it could matter somehow downstream.